### PR TITLE
Add CentOS 7 EOL to OS descriptions

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

Add descriptions about CentOS 7 EOL and CentOS Stream 8 EOL

**Note:** "CentOS 7" also exists in many other documents, which will be updated as needed later in another PR.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by JasonWu0506** [comment]
LGTM

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -34,7 +34,8 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
 
     > **注意：**
     >
-    > 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止，但 CentOS 将继续提供对 CentOS Stream 8 的支持。
+    > - 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止，但 CentOS 将继续提供对 CentOS Stream 8 的支持。
```
Comment (lilin90) [inline hardware-and-software-requirements.md:37]:
```suggestion
    > - 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止。[CentOS Stream 8 的上游支持](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) 已于 2024 年 5 月 31 日终止。CentOS 将继续提供对 CentOS Stream 9 的支持。
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -34,7 +34,8 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
 
     > **注意：**
     >
-    > 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止，但 CentOS 将继续提供对 CentOS Stream 8 的支持。
+    > - 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止。[CentOS Stream 8 的上游支持](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/) 已于 2024 年 5 月 31 日终止。CentOS 将继续提供对 CentOS Stream 9 的支持。
```
Comment (lilin90) [inline hardware-and-software-requirements.md:37]:
```suggestion
    > - 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止。[CentOS Stream 8 的上游支持](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)已于 2024 年 5 月 31 日终止。CentOS 将继续提供对 CentOS Stream 9 的支持。
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -34,7 +34,8 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
 
     > **注意：**
     >
-    > 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止，但 CentOS 将继续提供对 CentOS Stream 8 的支持。
+    > - 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止。[CentOS Stream 8 的上游支持](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)已于 2024 年 5 月 31 日终止。CentOS 将继续提供对 CentOS Stream 9 的支持。
+    > - 根据 [CentOS Linux EOL](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)，CentOS Linux 7 的上游支持于 2024 年 6 月 30 日终止。TiDB 将在 8.5 LTS 版本结束对 CentOS 7 的支持，建议使用 Rocky Linux 9.1 及以上的版本。
```
Comment (lilin90) [inline hardware-and-software-requirements.md:38]:
```suggestion
    > - 根据 [CentOS Linux EOL](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)，CentOS Linux 7 的上游支持于 2024 年 6 月 30 日终止。TiDB 将在 8.5 LTS 版本结束对 CentOS 7 的支持，建议使用 Rocky Linux 9.1 及以上的版本。
    > - 根据 [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/)，CentOS Linux 8 的上游支持已于 2021 年 12 月 31 日终止。[CentOS Stream 8 的上游支持](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/)已于 2024 年 5 月 31 日终止。CentOS 将继续提供对 CentOS Stream 9 的支持。
```

File: hardware-and-software-requirements.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
     |  操作系统   |   支持的 CPU 架构   |
     |   :---   |   :---   |
     | Red Hat Enterprise Linux 8.4 及以上的 8.x 版本  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
-    | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+    | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本 ⚠️</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```
Comment (lilin90) [inline hardware-and-software-requirements.md:27]:
```suggestion
    | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本（将在 8.5 LTS 版本结束支持）</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```

File: hardware-and-software-requirements.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ TiDB 作为一款开源一栈式实时 HTAP 数据库，可以很好地部署和
     |  操作系统   |   支持的 CPU 架构   |
     |   :---   |   :---   |
     | Red Hat Enterprise Linux 8.4 及以上的 8.x 版本  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
-    | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+    | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本（将在 8.5 LTS 版本结束支持）</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```
Comment (qiancai) [inline hardware-and-software-requirements.md:27]:
```suggestion
    | <ul><li>Red Hat Enterprise Linux 7.3 及以上的 7.x 版本</li><li>CentOS 7.3 及以上的 7.x 版本（将在 TiDB 8.5 LTS 版本结束支持）</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**